stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -559,7 +561,9 @@ void TaskGroupImpl::offer(AsyncTask *completedTask, AsyncContext *context) {
   assert(completedTask->groupChildFragment()->getGroup() == asAbstract(this));
   SWIFT_TASK_DEBUG_LOG("offer task %p to group %p", completedTask, this);
 
+#if !SWIFT_STDLIB_SINGLE_THREADED_RUNTIME
   mutex.lock(); // TODO: remove fragment lock, and use status for synchronization
```

Comment:
It would be nice to centralize these conditionals with small `lockMutex` and `unlockMutex` methods that are empty in the single-threaded runtime. That doesn't have to be done now, just something to consider if you find yourself bored one day.

Reply:
I've thought that every time I looked at this changeset, but needed a spare moment to actually make the change. Will do that now.

Reply:
Actually, `include/swift/Runtime/Mutex.h` has wrappers that will do the right thing for any given platform, and it already has a single-threaded stub implementation.

Reply:
Yeah, not sure what the deal is with that. I'd expect `Mutex.h` to be a better choice at least on Darwin, where it will use `os_unfair_lock`, but it could be that there are weird problems with it in this context.

Reply:
Does the C++11 standard (or later) allow for a conforming implementation that does not have `std::mutex`? I'd expect a single-threaded system to have the type but that using it would be a no-op.

Reply:
@grynspan Regardless of what the C++ standard has to say on the subject, there are definitely systems (that we care about) where we can't use `std::mutex`.